### PR TITLE
refactor: move util methods to util class

### LINKED ISSUE #16617: Refactoring: Extract re-bundle checker code to a standalone utils/helper, because it is reused for dev and prod bundle checks
Create BundleUtil class and move methods for different checks to the utils class to be used from development and production bundle tasks for checking if the bundle needs to be rebuilt.

### PR DESCRIPTION
**Pull request by caalador**
Move bundle verification methods
to utility class.

closes #16617

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java [hunk 1]
```diff
@@ -0,0 +1,616 @@
...
+public class BundleValidationUtil {
+
+    /**
+     * Check if jar bundle exists on given path.
+     *
+     * @param jarPath
+     *            JAR path where bunlde to check is located
+     * @return {@code true} if bundle stats.json is found
+     */
+    static boolean hasJarBundle(String jarPath) {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java:51]: package-private modifiers make this util methods inaccessible from `build-frontend` mojo where this check is supposed to happen to decide re-build or re-use the bundle. Maybe make it just public?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java [hunk 2]
```diff
@@ -0,0 +1,616 @@
...
+        final URL resource = TaskRunDevBundleBuild.class.getClassLoader()
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java:52]: Use `BundleValidationUtil` to get the classloader?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java (hunk 1 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java]: done

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java (hunk 2 above)
Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/BundleValidationUtil.java]: done

**Review by mshabarov — APPROVED** [review]

**Pull request merged**